indico/modules/users/controllers.py

```diff
@@ -202,19 +202,24 @@ def _check_access(self):
 class RHPersonalData(RHUserBase):
     allow_system_user = True
 
+    def get_titles(self):
+        return [{'name': t.name, 'title': t.title} for t in UserTitle if t != UserTitle.none]
+
     def _process(self):
-        titles = [{'name': t.name, 'title': t.title} for t in UserTitle if t != UserTitle.none]
+        titles = self.get_titles()
         user_values = UserPersonalDataSchema().dump(self.user)
         locked_fields = [] if session.user.is_admin else list(multipass.locked_fields)
         current_affiliation = None
         if self.user.affiliation_link:
             current_affiliation = AffiliationSchema().dump(self.user.affiliation_link)
         has_predefined_affiliations = Affiliation.query.filter(~Affiliation.is_deleted).has_rows()
+        extra_data = values_from_signal(signals.users.personal_data.send(self.user), as_list=True)
         return WPUserPersonalData.render_template('personal_data.html', 'personal_data', user=self.user,
                                                   titles=titles, user_values=user_values, locked_fields=locked_fields,
                                                   locked_field_message=multipass.locked_field_message,
                                                   current_affiliation=current_affiliation,
-                                                  has_predefined_affiliations=has_predefined_affiliations)
+                                                  has_predefined_affiliations=has_predefined_affiliations,
+                                                  extra_data=extra_data[0] if extra_data else None)
```

Comment:
this breaks if more than one plugin uses this signal. you may want to merge the data.

Reply:
I added the context when sending signal, so should not have multiple uses for the same class.

Reply:
i was thinking about cases where one has two plugins that both make use of the signal.. and more context does not help there

Reply:
Sorry, I still can't see the case you mentioned:
I expect the one using the signal to check if sender if RHPersonalData, then in the same Indico implementation, it means that connect is to do sth in the PD page,
then how the same Indico implementation can have another plugin working on that page without checking a signal connect already exists ?

Nevertheless, I will change code following your suggestion: chaining results and remove RH.

Reply:
My point is that Indico has no control over what plugins someone installs.

Let's say someone installs a "gender" plugin and someone else a "favorite color" plugin. So now you have two plugins, one of them returning `{'gender': 'male'}` and the other is returning `{'favorite_color': 'yellow'}`.

Reply:
Done as suggested: merge data and signal renamed to "extra_personal_data".